application/views/users/index.php

```diff
@@ -13,17 +15,20 @@
           <tbody>
               <?php foreach ($users as $user): ?>
                   <tr>
-                  <th scope="row"><?php echo $user["id"]; ?></th>
+                  <td><input type="checkbox" name="user_ids[]" value="<?php echo $user["id"]; ?>"></th>
+				  <th scope="row"><?php echo $user["id"]; ?></th>
                   <td><?php echo $user["first_name"]; ?></td>
                   <td><?php echo $user["last_name"]; ?></td>
                   <td>
-                      <a href="#"><button class="btn btn-primary btn-sm">View</button></a>
-					  <a href="<?php echo site_url("users/update/$user[id]"); ?>"><button class="btn btn-outline-primary btn-sm">Edit</button></a>
-					  <a onclick="return confirm('Are you sure to delete this user?')" href="<?php echo site_url("users/delete/$user[id]"); ?>"><button class="btn btn-sm">Delete</button></a>
+                      <button class="btn btn-primary btn-sm"><a href="<?php echo site_url("users/view"); ?>">View</a></button>
+					  <button class="btn btn-outline-primary btn-sm"><a href="<?php echo site_url("users/update/$user[id]"); ?>">Edit</a></button>
+					  <button class="btn btn-sm"><a onclick="return confirm('Are you sure to delete this user?')" href="<?php echo site_url("users/delete/$user[id]"); ?>">Delete</a></button>
```

Comment:
If I nest `<button>` with `<a>`, the button doesn't navigate to a linked page but the Controller calls the `all_delete` method which deletes all the selected users in the checkbox.  All the button in `form_open` & `form_close` seem under the influence of form's `action` attribute.
However, If I nest `<a>` with `<butoon>`, the problem is solved for the time being.